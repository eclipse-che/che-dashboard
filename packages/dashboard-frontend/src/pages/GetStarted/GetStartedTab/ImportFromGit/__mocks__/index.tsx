/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/GetStartedTab/ImportFromGit';

export class ImportFromGit extends React.PureComponent<Props> {
  public render() {
    const { hasSshKeys } = this.props;
    return (
      <div>
        <span data-testid="has-ssh-keys">{hasSshKeys ? 'true' : 'false'}</span>
      </div>
    );
  }
}
