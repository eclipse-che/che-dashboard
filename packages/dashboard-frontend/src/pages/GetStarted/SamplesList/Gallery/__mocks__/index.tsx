/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery';

export default class SamplesListGallery extends React.PureComponent<Props> {
  render() {
    const { onCardClick, metadataFiltered = [] } = this.props;
    return (
      <div data-testid="samples-list-gallery">
        <div>Samples List Gallery</div>
        {metadataFiltered.map(metadata => (
          <div key={metadata.links.v2} data-testid="sample-card">
            {metadata.displayName}

            <button
              onClick={() => {
                onCardClick(metadata);
              }}
            >
              Select Sample
            </button>
          </div>
        ))}
      </div>
    );
  }
}
