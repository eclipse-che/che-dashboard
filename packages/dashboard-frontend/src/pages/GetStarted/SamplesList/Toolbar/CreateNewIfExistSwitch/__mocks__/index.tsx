/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class CreateNewIfExistSwitch extends React.PureComponent<Props> {
  private isChecked = false;

  render(): React.ReactNode {
    return (
      <div data-testid="create-new-if-exist">
        {this.isChecked ? 'Create New If Existing On' : 'Create New If Existing Off'}
        <input
          data-testid="create-new-if-exist-switch"
          type="checkbox"
          onClick={() => {
            this.isChecked = !this.isChecked;
            this.props.onChange(this.isChecked);
          }}
          name="Toggle Create New If Existing"
        />
      </div>
    );
  }
}
