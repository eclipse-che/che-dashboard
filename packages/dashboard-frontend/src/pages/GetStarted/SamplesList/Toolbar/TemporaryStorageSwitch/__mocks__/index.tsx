/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export default class TemporaryStorageSwitch extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary } = this.props;
    return (
      <div>
        {isTemporary ? 'Temporary Storage On' : 'Temporary Storage Off'}
        <input
          type="checkbox"
          onClick={() => this.props.onChange(!isTemporary)}
          name="Toggle Temporary Storage"
        />
      </div>
    );
  }
}
