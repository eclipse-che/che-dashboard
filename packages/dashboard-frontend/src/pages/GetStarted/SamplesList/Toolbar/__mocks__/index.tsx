/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Toolbar';

export default class SamplesListToolbar extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary, onTemporaryStorageChange } = this.props;
    return (
      <div>
        <span>Samples List Toolbar</span>
        <span data-testid="isTemporary">{isTemporary ? 'true' : 'false'}</span>
        <button onClick={() => onTemporaryStorageChange(!isTemporary)}>Toggle isTemporary</button>
      </div>
    );
  }
}
