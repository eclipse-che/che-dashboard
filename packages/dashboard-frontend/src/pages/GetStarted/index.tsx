/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Divider, PageSection, PageSectionVariants, Title } from '@patternfly/react-core';
import { History } from 'history';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import Head from '@/components/Head';
import EditorSelector from '@/pages/GetStarted/EditorSelector';
import ImportFromGit from '@/pages/GetStarted/ImportFromGit';
import SamplesList from '@/pages/GetStarted/SamplesList';

type Props = MappedProps & {
  history: History;
};

export class GetStarted extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { history } = this.props;

    const title = 'Create Workspace';

    return (
      <React.Fragment>
        <Head pageName="Create Workspace" />

        <PageSection variant={PageSectionVariants.light}>
          <Title headingLevel={'h1'}>{title}</Title>
        </PageSection>
        <Divider />
        <PageSection variant={PageSectionVariants.default}>
          <PageSection variant={PageSectionVariants.light}>
            <EditorSelector />
          </PageSection>
          <PageSection
            variant={PageSectionVariants.light}
            style={{ marginTop: 'var(--pf-c-page__main-section--PaddingTop)' }}
          >
            <ImportFromGit history={history} />
          </PageSection>
          <PageSection
            variant={PageSectionVariants.light}
            style={{ marginTop: 'var(--pf-c-page__main-section--PaddingTop)' }}
          >
            <SamplesList history={history} />
          </PageSection>
        </PageSection>
      </React.Fragment>
    );
  }
}

const connector = connect();

type MappedProps = ConnectedProps<typeof connector>;
export default connector(GetStarted);
