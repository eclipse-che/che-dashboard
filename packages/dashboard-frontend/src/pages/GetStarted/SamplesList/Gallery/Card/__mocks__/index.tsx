/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card';

export class SampleCard extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { metadata, onClick } = this.props;

    return (
      <div data-testid="sample-card">
        <div data-testid="sample-name">{metadata.displayName}</div>
        <button onClick={() => onClick()}>Select Sample</button>
      </div>
    );
  }
}
