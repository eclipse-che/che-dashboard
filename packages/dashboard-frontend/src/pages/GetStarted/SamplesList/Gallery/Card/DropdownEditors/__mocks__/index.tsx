/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card/DropdownEditors';

export class DropdownEditors extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { editors, onEditorSelect } = this.props;
    return (
      <div data-testid="card-actions-dropdown">
        <div>spanDropdownEditors</div>
        {editors.map(editor => (
          <button
            data-testid="card-action"
            key={editor.id}
            onClick={() => onEditorSelect(editor.id)}
          >
            {editor.name}
          </button>
        ))}
        <button onClick={() => onEditorSelect(editors[0].id)}>Select Editor</button>
      </div>
    );
  }
}
