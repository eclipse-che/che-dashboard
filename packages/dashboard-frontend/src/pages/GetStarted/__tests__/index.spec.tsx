/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import GetStarted from '@/pages/GetStarted';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';

jest.mock('@/components/EditorSelector');
jest.mock('@/pages/GetStarted/SamplesList');
jest.mock('@/pages/GetStarted/ImportFromGit');

const { createSnapshot } = getComponentRenderer(getComponent);

describe('GetStarted', () => {
  test('snapshot', () => {
    const store = new FakeStoreBuilder().build();
    const snapshot = createSnapshot(store);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});

function getComponent(store: Store) {
  const history = createMemoryHistory();
  return (
    <Provider store={store}>
      <GetStarted history={history} />
    </Provider>
  );
}
