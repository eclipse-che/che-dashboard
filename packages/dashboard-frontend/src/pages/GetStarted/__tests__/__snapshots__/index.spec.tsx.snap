// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted snapshot 1`] = `
[
  <section
    className="pf-c-page__main-section pf-m-light"
  >
    <h1
      className="pf-c-title pf-m-2xl"
      data-ouia-component-id="OUIA-Generated-Title-1"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe={true}
    >
      Create Workspace
    </h1>
  </section>,
  <hr
    className="pf-c-divider"
  />,
  <section
    className="pf-c-page__main-section"
  >
    <div
      data-testid="editor-selector"
    >
      Editor Selector
      <button
        onClick={[Function]}
      >
        Select Editor
      </button>
    </div>
    <div
      className="spacer"
      data-testid="spacer"
    />
    <div
      data-testid="import-from-git"
    >
      <div>
        Import from Git
      </div>
      <div
        data-testid="editor-definition"
      />
      <div
        data-testid="editor-image"
      />
    </div>
    <div
      className="spacer"
      data-testid="spacer"
    />
    <div
      data-testid="samples-list"
    >
      <div
        data-testid="editor-definition"
      />
      <div
        data-testid="editor-image"
      />
      Samples List
    </div>
  </section>,
]
`;
